### PR TITLE
feat(core): use twig for item_disk

### PR DESCRIPTION
**Pull request by stonebuzz**
Use twig for ```Item_Disk``` (usefull for ```locked field```)
I take this opportunity to add the "Lock" tab

Before : 
image

After : 
image

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by orthagh — APPROVED** [review]
Is ok.

Better for main branch, no ?

**Comment by stonebuzz** [comment]
> Is ok.
> Better for main branch, no ?

I'd like to keep 10.0/bugfixes if possible because this PR shows ```LockedField``` if there are any (the old "showForm" doesn't)

**Comment by stonebuzz** [comment]
@orthagh, @trasher , @cedric-anne 

I am not satisfied with displaying the ```is_dynamic``` field (with ```generic_show_form.html.twig```)

It seems that ```Item_Disk``` is the only object that displays this field in its ```showForm```

Now other object displays this information but in different places

```Computer```
image

```Operating System```
image

I propose toremove the display of this field , then in another PR display this information as a badge (perhaps)

what do you think?

++

File: src/Item_Disk.php [hunk 1]
```diff
@@ -117,7 +119,7 @@ public function defineTabs($options = [])
      *     - items_id ID of the item for add process
      *
      * @return true if displayed  false if item not found or not right to display
-     **/
+    -     **/
```
Comment (cedric-anne) [inline src/Item_Disk.php:122]:
```suggestion
    **/
```

File: templates/components/form/item_disk.html.twig [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+   ) }}
+
+   {{ fields.textField(
+      'encryption_type',
+      item.fields['encryption_type'],
+      __('Encryption Type'),
+      field_options|merge(params)
+   ) }}
+
+{% endblock %}
```
Comment (cedric-anne) [inline templates/components/form/item_disk.html.twig:118]:
```suggestion
{% endblock %}

```

File: templates/generic_show_form.html.twig [hunk 3]
```diff
@@ -629,6 +629,12 @@
                         ) }}
                      {% endif %}
 
+                     {# display is_dynamic data without input to prevent "lock field" process to be avoid #}
+                     {% if item.isField('is_dynamic') %}
+                        {% set is_dynamic_value = call('Dropdown::getYesNo', [item.fields['is_dynamic']]) %}
```
Comment (cedric-anne) [inline templates/generic_show_form.html.twig:634]: The hook `autoinventory_information` should probably be called here.

Comment (stonebuzz) [reply templates/generic_show_form.html.twig:634]: is already called from this file (at end) =)

File: src/Item_Disk.php [hunk 4]
```diff
@@ -117,7 +119,7 @@ public function defineTabs($options = [])
      *     - items_id ID of the item for add process
      *
      * @return true if displayed  false if item not found or not right to display
-     **/
+    **/
```
Comment (cedric-anne) [inline src/Item_Disk.php:122]:
```suggestion
     **/
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]
:+1:

**Pull request merged**